projects/igniteui-angular/src/lib/directives/date-time-editor/date-time-editor.directive.ts

```diff
@@ -301,30 +327,28 @@ export class IgxDateTimeEditorDirective extends IgxMaskDirective implements OnCh
    *
    * @param datePart The optional DatePart to increment. Defaults to Date or Hours(when Date is absent from the inputFormat - ex:'HH:mm').
    */
-  public increment(datePart?: DatePart): void {
+  public increment(datePart?: DatePart, spinDeltas?: DatePartDeltas): void {
+    const deltas = Object.assign({}, this.datePartDeltas, spinDeltas);
     const targetDatePart = this.targetDatePart;
     if (!targetDatePart) {
       return;
     }
-    const newValue = datePart
-      ? this.spinValue(datePart, 1)
-      : this.spinValue(targetDatePart, 1);
-    this.updateValue(newValue ? newValue : new Date());
+    const newValue = this.trySpinValue(datePart, targetDatePart, deltas);
```

Comment:
And when the part is determined, there should be no need to pass two parts to this method